manifests/init.pp

```diff
@@ -250,12 +252,19 @@
   Boolean $restart_sssd = true,
   Optional[Stdlib::Absolutepath] $service_environment_file = undef,
   Enum['standalone', 'clustered'] $operating_mode = 'standalone',
+  Optional[Array] $tech_preview_features = undef,
 ) {
 
   if ! $facts['os']['family'] in ['RedHat','Debian'] {
     fail("Unsupported osfamily: ${facts['os']['family']}, module ${module_name} only support osfamilies Debian and Redhat")
   }
 
+  $profile_properties_ensure = $tech_preview_features ? {
```

Comment:
If this becomes just empty array can do something like `if empty($tech_preview_features) { `